winrm/client.go

```diff
@@ -23,21 +29,38 @@ type Client struct {
 
 // NewClient will create a new remote client on url, connecting with user and password
 // This function doesn't connect (connection happens only when CreateShell is called)
-func NewClient(endpoint *Endpoint, user, password string) (client *Client, err error) {
+func NewClient(endpoint *Endpoint, user, password string, auth AuthType) (client *Client, err error) {
```

Comment:
Actually, certificate authentication and basic authentication in the case of WinRM, are mutually exclusive. You can have both enabled, but you can use only one at a time.

Certificate authentication requires you to create a mapping between a local account and a certificate. In which case, when authenticating using a x509 cert, you gain access to the account to which the certificate is mapped.

See:

https://msdn.microsoft.com/en-us/library/aa384295%28v=vs.85%29.aspx

for more information.